thpool.c

```diff
@@ -89,7 +89,7 @@ typedef struct thpool_{
 	volatile int num_threads_working;    /* threads currently working */
 	pthread_mutex_t  thcount_lock;       /* used for thread count etc */
 	pthread_cond_t  threads_all_idle;    /* signal to thpool_wait     */
-	jobqueue  jobqueue;                  /* job queue                 */
```

Comment:
I guess this is the main issue with C++? Isn't there another way to silence the warning without changing the variable names? (never used C++)

I'm just hesitant, since renaming to `job_queue` breaks naming convention with other functions

Reply:
I think of a simple way to solve it by using extern "C" to avoid problems in C++.

Reply:
But there is also a problem with this code style. The struct name should not be consistent with the variable name.

Reply:
Using jobqueue_t may be a better choice.